src/plugins/fs.c

```diff
@@ -373,6 +491,33 @@ static gboolean wipe_fs (const gchar *device, const gchar *fs_type, GError **err
         return FALSE;
     }
 
+    if (wipe_all) {
+        gboolean has_fat = TRUE;
+        has_fat = has_fs (device, "vfat", error);
```

Comment:
And it looks like opening and probing the device over and over by calling this function is not necessary when the ``wipe_fs`` function has the device opened and is probing it.